php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1045 @@
+  private final TreeFactory f;
+
+  public NewPHPGrammar(GrammarBuilder<InternalSyntaxToken> b, TreeFactory f) {
+    this.b = b;
+    this.f = f;
+  }
+
+  /**
+   * [ START ] Declaration
+   */
+
+  public VariableDeclarationTree MEMBER_CONST_DECLARATION() {
+    return b.<VariableDeclarationTree>nonterminal(PHPLexicalGrammar.MEMBER_CONST_DECLARATION)
+        .is(f.memberConstDeclaration(
+            b.token(PHPLexicalGrammar.IDENTIFIER),
+            b.optional(f.newTuple18(b.token(EQU), STATIC_SCALAR()))
+        ));
+  }
+  
+  public VariableDeclarationTree VARIABLE_DECLARATION() {
+    return b.<VariableDeclarationTree>nonterminal(PHPLexicalGrammar.VARIABLE_DECLARATION)
+        .is(f.variableDeclaration(
+            b.token(PHPLexicalGrammar.REGULAR_VAR_IDENTIFIER),
+            b.optional(f.newTuple96(b.token(EQU), STATIC_SCALAR()))
+        ));
+  }
+
+  public NamespaceNameTree NAMESPACE_NAME() {
+    return b.<NamespaceNameTree>nonterminal(PHPLexicalGrammar.NAMESPACE_NAME)
+        .is(b.firstOf(
+          f.namespaceName(
+            b.optional(b.token(PHPPunctuator.NS_SEPARATOR)),
+            b.zeroOrMore(f.newTuple4(
+              b.firstOf(b.token(PHPLexicalGrammar.IDENTIFIER)),
+              b.token(PHPPunctuator.NS_SEPARATOR)
+            )),
+            b.token(PHPLexicalGrammar.IDENTIFIER)),
+          f.namespaceName(
+              b.token(PHPKeyword.NAMESPACE),
+              b.token(PHPPunctuator.NS_SEPARATOR),
+              b.zeroOrMore(f.newTuple6(
+                  b.firstOf(b.token(PHPLexicalGrammar.IDENTIFIER)),
+                  b.token(PHPPunctuator.NS_SEPARATOR)
+              )),
+              b.token(PHPLexicalGrammar.IDENTIFIER)))
+    );
+  }
+  
+  public UseClauseTree USE_CLAUSE() {
+    return b.<UseClauseTree>nonterminal(PHPLexicalGrammar.USE_DECLARATION).is(
+      f.useClause(
+          NAMESPACE_NAME(),
+          b.optional(
+              f.newTuple91(
+                  b.token(PHPKeyword.AS),
+                  b.token(PHPLexicalGrammar.IDENTIFIER)))));
+  }
+  
+  public ClassPropertyDeclarationTree CLASS_CONSTANT_DECLARATION() {
+    return b.<ClassPropertyDeclarationTree>nonterminal(PHPLexicalGrammar.CLASS_CONSTANT_DECLARATION).is(
+      f.classConstantDeclaration(
+        b.token(PHPKeyword.CONST),
+        MEMBER_CONST_DECLARATION(),
+        b.zeroOrMore(f.newTuple97(b.token(COMMA), MEMBER_CONST_DECLARATION())),
+        EOS()));
+  }
+  
+  public ClassPropertyDeclarationTree CLASS_VARIABLE_DECLARATION() {
+    return b.<ClassPropertyDeclarationTree>nonterminal(PHPLexicalGrammar.CLASS_VARIABLE_DECLARATION).is(
+      f.classVariableDeclaration(
+        b.firstOf(
+          f.singleToken(b.token(PHPKeyword.VAR)),
+          b.oneOrMore(MEMBER_MODIFIER())),
+        VARIABLE_DECLARATION(),
+        b.zeroOrMore(f.newTuple95(b.token(COMMA), VARIABLE_DECLARATION())),
+        EOS()));
+  }
+  
+  public SyntaxToken MEMBER_MODIFIER() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)